catalyst-gateway/bin/src/rbac/indexing_context.rs

```diff
@@ -0,0 +1,125 @@
+//! A RBAC context used during indexing.
+
+use std::collections::{HashMap, HashSet};
+
+use cardano_blockchain_types::{Slot, StakeAddress, TransactionId, TxnIndex};
+use catalyst_types::catalyst_id::CatalystId;
+use ed25519_dalek::VerifyingKey;
+
+use crate::db::index::queries::rbac::get_rbac_registrations::Query as RbacQuery;
+
+/// A RBAC context used during indexing.
+///
+/// During indexing the information is written to the database only after processing the
+/// whole block. If there are multiple RBAC registrations in one block, then when
+/// validating subsequent transactions we wouldn't be able to find information about the
+/// previous ones in the database. This context is used to hold such information during
+/// block processing in order to mitigate that issue.
+pub struct RbacIndexingContext {
```

Comment:
I think it make sense to emphisize that it is a per block indexing context, not a general context for indexing.
```suggestion
pub struct RbacBlockIndexingContext {
```